tests/datasets/test_geo.py

```diff
@@ -178,6 +178,26 @@ def sentinel(self, request: SubRequest) -> Sentinel2:
         cache = request.param[1]
         return Sentinel2(root, bands=bands, transforms=transforms, cache=cache)
 
+    def test_init_with_list_of_files(self) -> None:
```

Comment:
Can we do a pytest.parametrize and test all of the following:

* single dir (str)
* multiple dirs (list[str])
* single file (str)
* multiple files (list[str])
* combination of dirs and files (list[str])

Should also rename the test to something more generic. We should either move this to TestGeoDataset or also add something similar to TestVectorDataset.

Reply:
Good idea. If you want to take a crack at testing this and codecov, I can take a look at the VectorDataset support?